code/01_spatial_overlay.R

```diff
@@ -142,14 +136,30 @@ calc_geoweights <- function(data_source = 'era5',  input_polygons, polygon_id, w
     # Weight in pixel = w_area * weight
     w_merged[, weight := weight * w_area]
     
+    # Create column that determines if entire polygon has a weight == 0
+    zero_polys <- data.frame(w_merged) %>%
+      group_by(poly_id) %>%
+      summarise(sum_weight = sum(weight)) %>%
+      ungroup() %>%
+      filter(sum_weight == 0) %>%
+      dplyr::select(poly_id) %>%
+      distinct()
+    
+    if(nrow(zero_polys > 0)) {
```

Comment:
This makes me think of a possible change to the NAs too. What if we copy this & adjust for na polys so adding something like this at line 160:
```
if(nrow(na_polys > 0)) {

      warning(crayon::red("Warning: some polygons fall outside the weights extent; area-weights will be returned"))

    }
```

And then we can delete the part that checks extents and outputs a warning (e.g. lines 116-125. What do you think?

Reply:
See most recent update!